From e62d81870d178245eb37b7151ed7128a28c21c0f Mon Sep 17 00:00:00 2001
From: RickyWu-wiwynn <ricky_cx_wu@wiwynn.com>
Date: Fri, 1 Nov 2024 13:54:38 +0800
Subject: [PATCH 1002/1016] arm: dts: aspeed: yosemite4: Revise i2c duty cycle

Revise duty cycle SMB9, SMB10, SMB11, and SMB16 to 40:60 to meet 400kHz-i2c clock low time spec (> 1.3 us).
- Haven't contributed because it is depend on the following patch:
https://lore.kernel.org/all/47e7eb15-e38d-ead3-de84-b7454e2c6eb8@gmail.com/
---
 arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-yosemite4.dts | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-yosemite4.dts b/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-yosemite4.dts
index 1fbe050b428b..57c72b08adab 100644
--- a/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-yosemite4.dts
+++ b/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-yosemite4.dts
@@ -583,6 +583,8 @@ &i2c8 {
 	#size-cells = <0>;
 	status = "okay";
 	bus-frequency = <400000>;
+	i2c-clk-high-min-percent = <40>;
+
 	i2c-mux@70 {
 		compatible = "nxp,pca9544";
 		reg = <0x70>;
@@ -705,6 +707,8 @@ &i2c9 {
 	#size-cells = <0>;
 	status = "okay";
 	bus-frequency = <400000>;
+	i2c-clk-high-min-percent = <40>;
+
 	i2c-mux@71 {
 		compatible = "nxp,pca9544";
 		reg = <0x71>;
@@ -827,6 +831,8 @@ &i2c10 {
 	#size-cells = <0>;
 	status = "okay";
 	bus-frequency = <400000>;
+	i2c-clk-high-min-percent = <40>;
+
 	i2c-mux@74 {
 		compatible = "nxp,pca9544";
 		reg = <0x74>;
@@ -1283,6 +1289,7 @@ &i2c15 {
 	status = "okay";
 	multi-master;
 	bus-frequency = <400000>;
+	i2c-clk-high-min-percent = <40>;
 
 	mctp@10 {
 		compatible = "mctp-i2c-controller";
-- 
2.47.3

